Libraries/HyperV.psm1

```diff
@@ -258,13 +272,39 @@ Function DeleteHyperVGroup([string]$HyperVGroupName, [string]$HyperVHost) {
         LogMsg "Stop-VM -Name $($_.Name) -Force -TurnOff "
         $vm = $_
         Stop-VM -Name $vm.Name -Force -TurnOff -ComputerName $HyperVHost
+        Remove-VMSnapshot -VMName $vm.Name -ComputerName $HyperVHost `
+            -IncludeAllChildCheckpoints -Confirm:$false
+        if (!$?) {
+            LogErr ("Failed to remove snapshots for VM {0}" -f @($vm.Name))
+            return $false
+        }
+        Wait-VMStatus -VMName $vm.Name -VMStatus "Operating Normally" -RetryInterval 2 `

```

Comment:
I don't think this wait-vmstatus is required, you force a vm stop above so there is no other state in which the VM can be.
and 'operating normally' appears independent of VM state.

Reply:
This is part of Adi Vladu's PR with improvements to the vhd cleanup